### PR TITLE
MINOR Consolidate junit-platform.properties

### PR DESCRIPTION
**Pull request by mumrah**
Now that we have ":test-common" we can fix the underlying problem from KAFKA-17121. We should only have a single junit-platform.properties in our runtime classpath since JUnit does not guarantee which one will be loaded (https://github.com/junit-team/junit5/issues/2794).

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

If I recall correctly, junit-platform.properties is used to generate the correct test naming, such as quorum=kraft and quorum=zk, for the older test infrastructure. In short, junit-platform.properties can be removed once ZK is completely removed from the tests.

**Comment by mumrah** [comment]
Same four tests are failing on both Java version:

```
Found 4 test failures:
FAILED ❌ AppInfoParserTest > testUnregisterAppInfoUnregistersMetrics()
FAILED ❌ AppInfoParserTest > testRegisterAppInfoRegistersMetrics()
FAILED ❌ CommonNameLoggingTrustManagerFactoryWrapperTest > testCommonNameLoggingTrustManagerWithExpiredEndCert()
FAILED ❌ CommonNameLoggingTrustManagerFactoryWrapperTest > testCommonNameLoggingTrustManagerWithExpiredEndCertWithCA()
```

🤔 

Edit: Looks like some clients tests make assumptions about the classpath. I'll revert the changes to `:clients` module.

**Comment by mumrah** [comment]
> If I recall correctly, junit-platform.properties is used to generate the correct test naming
Yes, I think it's used in general for parameterized tests, not just ZK/KRaft. 

This file also lets us register JUnit plugins easily. This may be coming soon in a future PR :)

**Comment by chia7712** [comment]
> This file also lets us register JUnit plugins easily. This may be coming soon in a future PR :)

I just open a jira to remove the file :smile: 

Let me close it

File: build.gradle [hunk 1]
```diff
@@ -1897,7 +1897,6 @@ project(':clients') {
 
     testRuntimeOnly libs.jacksonDatabind
     testRuntimeOnly libs.jacksonJDK8Datatypes
-    testRuntimeOnly runtimeTestLibs
```
Comment (chia7712) [inline build.gradle]: `clients` needs `runtimeTestLibs` to run `CommonNameLoggingTrustManagerFactoryWrapperTest` with logging.

**Review by chia7712 — APPROVED** [review]

**Pull request merged**